contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,367 @@
+BestEffort cgroup sandbox memory limits in an effort to satisfy the
+specified level of overcommit.
+
+* The flag will accept values in a range from 0-100.
+* A value of 100 will instruct the `kubelet` to adjust the Burstable and
+BestEffort cgroup sandboxes to allow maximal overcommit.
+* A value of 0 will instruct the `kubelet` to adjust the Burstable and
+BestEffort cgroup to restrict memory overcommit by inducing memory
+pressure and reclaim if needed at the QoS level cgroup tiers before
+inducing pressure at the `ROOT` cgroup.
+
+The cgroup sandboxes are adjusted as follows:
+
+```
+ROOT/Burstable/memory.limit_in_bytes = 
+    Node.Allocatable - {(summation of memory requests of `Guaranteed` pods)*(1-qmo/100)}
+ROOT/BestEffort/memory.limit_in_bytes = 
+    Node.Allocatable - {(summation of memory requests of all `Guaranteed` and `Burstable` pods)*(1-qmo/100)}
+```
+
+Each time a pod is admitted to the `kubelet`, the `kubelet` will increment
+the QoS level cgroup memory limits.  In order to safely reduce memory limits, 
+the `kubelet` will run a separate control loop that attempts to reduce memory
+limits to the current desired state.  The kernel prevents the limit from being
+reduced below its current usage rather than inducing OOM.  If the `kubelet` 
+detects the current usage is greater than the desired limit, it will attempt to
+induce memory pressure and associated reclaim at the QoS level cgroups tiers
+by gradually reducing the memory limit value to meet the desired state.  This
+increases the risk of inducing OOM on `BestEffort` workloads in favor of
+increasing memory resource guarantees for `Burstable` and `Guaranteed`
+workloads.
+
+## Memory backed volumes
+
+All memory backed volumes are removed when a pod reaches a terminal state.
+
+This ensures resource consumption associated with those volumes are not
+incorrectly transferred back to the `kubelet` and/or node.
+
+## Rollout Plan
 
-For the value specified for the --qos-memory-overcommitment flag
-`qmo⇒ qos-memory-overcommitment`
+### Kubernetes 1.5
 
-Currently the Kubelet highly prioritizes resource utilization and thus allows BE pods to use as much resources as they want. And in case of OOM the BE pods are first to be killed. We follow this policy as G pods often don't use the amount of resource request they specify. By overcommiting the node the BE pods are able to utilize these left over resources. And in case of OOM the BE pods are evicted by the eviciton manager. But there is some latency involved in the pod eviction process which can be a cause of concern in latency-sensitive servers. On such servers we would want to avoid OOM conditions on the node. Pod level cgroups allow us to restrict the amount of available resources to the BE pods. So reserving the requested resources for the G and Bu pods would allow us to avoid invoking the OOM killer.
+The support for the unified cgroup hierarchy is experimental.
```

Comment:
I do not like the term `unified hierarchy`. As I mentioned earlier, it could mean several things. Why not name this feature, `QoS cgroups`?